app/services/search/query_builders/classified_listing.rb

```diff
@@ -0,0 +1,97 @@
+module Search
+  module QueryBuilders
+    class ClassifiedListing
+      TERM_KEYS = %i[
+        category
+        contact_via_connect
+        location
+        slug
+        tags
+        title
+        user_id
+      ].freeze
+
+      RANGE_KEYS = %i[
+        bumped_at
+        expires_at
+      ].freeze
+
+      QUERY_KEYS = %i[
+        classified_listing_search
+      ].freeze
+
+      DEFAULT_PARAMS = {
+        sort_by: "bumped_at",
+        sort_direction: "desc",
+        size: 0
+      }.freeze
+
+      attr_accessor :params, :body
+
+      def initialize(params)
+        @params = params.deep_symbolize_keys
+        build_body
+      end
+
+      def as_hash
+        @body
+      end
+
+      private
+
+      def build_body
+        @body = ActiveSupport::HashWithIndifferentAccess.new
+        build_queries
+        add_sort
+        # By default we will return 0 documents if size is not specified
+        @body[:size] = @params[:size] || DEFAULT_PARAMS[:size]
+      end
+
+      def build_queries
+        @body[:query] = {}
+        @body[:query][:bool] = { filter: filter_conditions }
```

Comment:
You likely will also want to check if filter conditions are present. I do not in `ChatChannelMemberships` bc we are always going to have a filter with the `user_id` that we pass in. Here, if there is a chance of no filter bring present we will want to check for it.